data/scripts.js

```diff
@@ -660,8 +660,8 @@ let BattleScripts = {
 		}
 
 		if (target) {
-			/**@type {?boolean | number} */
-			let didSomething = false;
+			/**@type {?boolean | number | undefined} */
```

Comment:
Do `null` and `undefined` mean different things here?

Reply:
Unfortunately yes, because some of the things that a move can try to do can give back a value of `null` which means "the attempted effect failed, but had its own failure message so don't send another one" in this context, while `undefined` simply means that the value hasn't been overwritten by anything, which indicates that the move didn't actually try to _do_ anything, which isn't supposed to count as failing; see the comment on line 776.

Reply:
That makes sense. `undefined` is a supported return value with the meaning "nothing unusual happened". But in this case, is it necessary to distinguish it from `true`?

Reply:
Well, it needs to be considered the same as true later, but it needs to be distinguishable before evaluating whether the move failed or not because all the future possible assignments to `didSomething` need to not overwrite its value if `didSomething` was already set to true by another effect, as could be the case if Toxic Thread successfully poisons the target but does not change its speed.

Reply:
Actually Toxic Thread might be the only standard move that actually does more than one of these things, if I remember correctly from when I originally implemented this.

Reply:
Oh, I see. That makes sense.